spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncherRunner.java

```diff
@@ -18,45 +18,45 @@
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+
 import org.springframework.beans.factory.InitializingBean;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.ApplicationArguments;
 import org.springframework.boot.ApplicationRunner;
-import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.stereotype.Component;
 import org.springframework.util.Assert;
 import org.springframework.util.StringUtils;
 
 /**
+ * Spring boot {@link ApplicationRunner} that triggers {@link ModuleLauncher} to launch the modules.
+ *
  * @author Marius Bogoevici
+ * @author Ilayaperumal Gopinathan
  */
 @Component
-@ConfigurationProperties
-// TODO: use a prefix
+@EnableConfigurationProperties(ModuleLauncherProperties.class)
 public class ModuleLauncherRunner implements ApplicationRunner, InitializingBean {
 
 	private final static Log log = LogFactory.getLog(ModuleLauncherRunner.class);
 
 	@Autowired
-	private ModuleLauncher moduleLauncher;
-
-	private String[] modules;
+	private ModuleLauncherProperties moduleLauncherProperties;
 
-	public void setModules(String[] modules) {
-		this.modules = modules;
-	}
+	@Autowired
+	private ModuleLauncher moduleLauncher;
 
 	@Override
 	public void afterPropertiesSet() throws Exception {
-		Assert.notEmpty(this.modules, "A list of modules must be specified");
+		Assert.notEmpty(moduleLauncherProperties.getModules(), "A list of modules must be specified");
```

Comment:
That would be nice. But, do we really want to have `spring-boot-starter-validation` dependency for `spring-cloud-stream` or at least for `module-launcher`? I see `spring-boot-starter-validation` defaults to `hibernate-validator`.